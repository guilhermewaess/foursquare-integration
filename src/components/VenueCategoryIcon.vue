<template>
  <div class="primary icon-container">
    <v-tooltip right
               color="primary">
      <v-card-media :src="getIconSrc()"
                    height="100px"
                    class="border-50"
                    contain
                    slot="activator">
      </v-card-media>
      <span>{{category.pluralName}}</span>
    </v-tooltip>
  </div>
</template>

<script>
export default {
  name: 'VenueCategoryIcon',
  props: {
    categories: {
      required: true,
      type: Array,
    },
  },
  methods: {
    getIconSrc() {
      return `${this.category.icon.prefix}32${this.category.icon.suffix}`;
    },
  },
  computed: {
    category() {
      return this.categories.find(category => category.primary);
    },
  },
};
</script>

<style lang="scss">
.icon-container {
  position: relative;

  .border-50 {
    border-radius: 50%;
  }
}
</style>
