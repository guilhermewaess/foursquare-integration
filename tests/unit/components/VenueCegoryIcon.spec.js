import { shallowMount } from '@vue/test-utils';
import VenueCategoryIcon from '@/components/VenueCategoryIcon.vue';

describe('VenueInformations', () => {
  let categories;
  let component;
  beforeEach(() => {
    categories = [
      {
        id: '4bf58dd8d48988d1fd941735',
        name: 'Shopping Mall',
        pluralName: 'Shopping Malls',
        shortName: 'Mall',
        icon: {
          prefix: 'prefix',
          suffix: 'suffix',
        },
        primary: false,
      },
      {
        id: '4bf58dd8d48988d181941735',
        name: 'Museum',
        pluralName: 'Museums',
        shortName: 'Museum',
        icon: {
          prefix: 'prefix2',
          suffix: 'suffix2',
        },
        primary: true,
      },
    ];
    component = shallowMount(VenueCategoryIcon, {
      propsData: { categories },
      stubs: ['v-tooltip', 'v-card-media'],
    });
  });
  it('should have name', () => {
    expect(component.name()).toEqual('VenueCategoryIcon');
  });
  describe('snapshot', () => {
    it('should render', () => {
      expect(component.element).toMatchSnapshot();
    });
  });
  describe('methods', () => {
    it('getIconSrc should return interpolation of icon prefix 32 and suffix', () => {
      const expected = `${categories[1].icon.prefix}32${categories[1].icon.suffix}`;
      expect(component.vm.getIconSrc()).toEqual(expected);
    });
  });
  describe('computed', () => {
    it('category should return the primary category', () => {
      expect(component.vm.category).toEqual(categories[1]);
    });
  });
});
