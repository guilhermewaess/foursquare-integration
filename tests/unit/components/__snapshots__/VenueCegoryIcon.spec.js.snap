// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VenueInformations snapshot should render 1`] = `
<div
  class="primary icon-container"
>
  <v-tooltip-stub
    color="primary"
    right=""
  >
     
    <div>
      Museums
    </div>
  </v-tooltip-stub>
</div>
`;
